cms/djangoapps/api/docs/decisions/001-course-apps.rst

```diff
@@ -0,0 +1,222 @@
+Course Apps API
+_______________
+
+Status
+======
+Proposal
+
+Context
+=======
+
+The new `Course Authoring MFE`_ includes a new UX called "Pages and Resources"
+for configuring different aspects of the course experience such as progress,
+wiki, teams, discussions, etc. all from one place.
+
+Currently, the code for these aspects of the course experience is intermingled
+with the core platform, however, it would be useful to have a way to
+automatically discover the data for driving this page instead of hardcoding it
+such that in the future existing functionality can be split out into an external
+plugin and so that future third-party plugins can also show up here.
+
+This ADR describes how we can use the existing plugin architecture of Open edX
+to automatically discover such apps and expose them via an API. This API will
+be used to list the installed apps that are available for a course and to
+enable/disable these apps using the API.
+
+.. _Course Authoring MFE: https://github.com/edx/frontend-app-course-authoring/
+
+
+Decision
+========
+
+We propose to call such individual course features "Course Apps". They can be
+introduced as a new type of Open edX plugin. Any functionality that can be
+enabled or disabled at a course-level and can be configured by instructors is
+a candidate for being bundled as a Course App.
+
+To be able to show these Course Apps to course admins, they will need to provide
+some bits of metadata, such as a name, a description etc. Additionally we will
+need a common interface for such assps so they can be enabled/disabled
+uniformly.
```

Comment:
What does `uniformly` means here? Can you add a little detail on what it means for these apps to be enabled/disabled uniformly?